libraries/AP_ExternalAHRS/AP_ExternalAHRS_GSOF.cpp

```diff
@@ -0,0 +1,362 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//   Usage in SITL with hardware for debugging:
+//     $ sim_vehicle.py -v Plane --console --map -DG
+//     param set AHRS_EKF_TYPE 11
+//     param set EAHRS_TYPE 6
+//     For EAHRS as a GPS:
+//     param set GPS1_TYPE 21
+//   Configure GSOF 49,50,70 on UDP port 44444, "UDP Mode" and "UDP Broadcast Transmit"
+//   Consider setting EK3_SRC1_YAW to 2 on the bench...
+
+// Usage with NET parameters and ethernet in SITL with hardware:
+//     param set NET_ENABLE 1
+//     param set NET_P1_TYPE 2
+//     # Set up AHRS input
+//     param set NET_P1_PROTOCOL 36
+//     param set SIM_GPS1_TYPE 0
+//     param set NET_P1_PORT 44444
+//     param set EAHRS_TYPE 6
+//
+// Usage with serial in SITL with hardware:
+//     $ sim_vehicle.py -v Plane -A "--serial3=uart:/dev/ttyUSB0" --console --map -DG
+//     param set SERIAL3_PROTOCOL 36
+//     param set SERIAL3_BAUD 115200
+//     param set EAHRS_TYPE 6
+//     # ensure NET_* if off if you were using ethernet.
+//     # To enable the EAHRS to provide GPS:
+//     param set GPS2_TYPE 21
+
+// On most hardware, you must enable EAHRS:
+//     ./waf configure --board Pixhawk6X --enable-AHRS_EXT
+
+// GPS ride-along with EARHS as the 2nd GPS
+//      param set GPS_AUTO_SWITCH 0
+//      param set GPS2_TYPE 21
+//      param set GPS_PRIMARY 0
+
+#define ALLOW_DOUBLE_MATH_FUNCTIONS
```

Comment:
Somehow this causes some builds to fail even though the same pattern is used in other code.